core/trino-main/src/main/java/io/trino/operator/window/WindowPartition.java

```diff
@@ -183,6 +242,23 @@ public void processNextRow(PageBuilder pageBuilder)
     {
         checkState(hasNext(), "No more rows in partition");
 
+        // check for new peer group
+        if (currentPosition == peerGroupEnd) {
+            updatePeerGroup();
+        }
+
+        if (machine.isPresent()) {
+            processRowWithPatternMatching(pageBuilder);
+        }
+        else {
+            processRegularWindowRow(pageBuilder);
+        }
+
```

Comment:
What's the benefit of having both of these in the same class if we're going to switch the behavior at this level? What's common between them? We may want to consider splitting them up and encapsulating the common behaviors in some shared component.